docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+            path_recorded = ""
+            for __, dirnames, filename in os.walk(dir_path):
+                # This is done to prevent writing directories below  the sub-directories
+                # i.e. we don't need to link any directories (if they are in future) that are below external with the common toctree
+                # the way this `if` statements works is: os.walk will keep iterating through a directory recursively (going into sub-sub-directories), we just need its first output
+                # of the directories within the root, after that it goes through the sub-sub-directories and lists their directories which we don't want,
+                # meanwhile path_main_file is constant, thus we can skip recording sub-subdirectories if we tell it that, this path has alreasy been recorded.
+                # remember path_main_file is remaining constant for one dir_path
+                if path_recorded == path_main_file:
+                    continue
+                # this counts the number of sub-directories each directory has
+                # true only for `common` (as of April 24, 2020)
+                if len(dirnames) > 0:
+                    for dirname in dirnames:
+                        if "__" in dirname:  # to prevent __pychache__ folders from being considered
+                            continue
+                        dir_main = dirname + "_main"  # external_main
+                        path_recorded = path_main_file
+                        lines = f"\n\t {dirname}/{dir_main}"
+                        outlines += lines
+            fh.write(outlines)
+    # Opens each DirectoryName_main.rst and add the name of the scripts in that directory
+    with open(path_main_file, "a") as fh:
+        fname = os.path.basename(fpath)
+        # extract the "kmers" of kmers.py
+        basename = os.path.splitext(fname)[0]
+        fh.write(f"\n\t {basename}")
+
+
+def link_write_usage():
+    """
+    Creates usage.rst and writes the directories in its toctree
+    """
+    if not os.path.exists(path_usage):
+        with open(path_usage, "w") as fh:
+            write_text = "\n".join(["="*len("usage"), "Usage", "="*len("usage"), " ", ".. toctree::",
+                                    "\t :maxdepth: 3", "\t :caption: Table of Contents", " "])
+            fh.write(write_text)
+            # This can also be done using next() -> https://stackoverflow.com/a/142535/12671809 ,
+            # but reduces readability. What to do ??
+            for dirname in os.listdir(path_scripts):
+                path_dirname = os.path.join(path_scripts, dirname)
+                if os.path.isdir(path_dirname):
+                    dirname_rst = dirname + "_main"
+                    fh.write(f"\n\t {dirname}/{dirname_rst}")
+
+
+def remove_existing_docs():
+    """
+    Removes the exists documentatio in `scripts` directory
+    """
+    if os.path.exists(path_scripts):
+        shutil.rmtree(path_scripts)
```

Comment:
Similar to comments above. `path` is ambiguous. specifying directory or file path helps understand what this is supposed to handle. Again, good to have this take what it is manipulating as an argument for the function. Manipulating variables outside of the context of the function requires reading through other sections of the script, where really, it would be nice to understand a function as defined within its own context.

```suggestion
def remove_existing_docs(scripts_dirpath):
    """
    Removes the existing documentation in `scripts` directory
    """
    if os.path.exists(scripts_dirpath):
        shutil.rmtree(scripts_dirpath)
```

Reply:
I had earlier defined `path_scripts` as global variable. I have moved them inside respective functions now.
Also `path_scripts` is now renamed to `scripts_dirpath`